### DISCUSSION TITLE
rspamd/razor does not work

### DISCUSSION THREAD

**Discussion by feiaweng2** [rspamd/rspamd#4237] [Q&A]
Hi community,

I have installed razor as described in documentation https://rspamd.com/doc/modules/external_services.html
In the rspamd.log I see only these lines:
#13759(normal) <1893ba>; lua; razor.lua:121: razor: failed to scan, maximum retransmits exceed
2022-08-05 00:04:14 #13759(normal) <1893ba>; lua; common.lua:110: razor: result - FAILED with error: "failed to scan and retransmits exceed - score: 0"
I have reinstalled razor. razor-check itself is working. But I guess there is a problem in communication between rspamd and razor.
I have no idea anymore.
Can anyone help me?

thanks in advance
feiaweng2

**Comment by segmentierungsfehler** [comment]
Hi

Did you already solve your problem? If not, how did you configure rspamd to use razor?
Here is my razor part of /etc/rspamd/local.d/external_services.conf (rspamd 3.2):
```
razor {
  symbol="RAZOR"
  servers = "127.0.0.1:11342"
  timeout = 10.0
  cache_expire = 7200
  retransmits = 1
}
```
And on localhost tcp port 11342 there is razor waiting for a connection. 

Bye

**Comment by feiaweng2** [comment]
Hi,

 

yes I have still the same problem.

My  /etc/rspamd/local.d/external_services.conf (rspamd 3.2) looks like as
yours:

####################################################################

razor {

  # default razor settings

  symbol="RAZOR"

  servers = "127.0.0.1:11342"

timeout = 10.0

  cache_expire = 7200

  retransmits = 1

}

####################################################################

 

 

If I feed razor-check with an email, it works fine.

razor-check -d "Email"

But rspamd is not able to work with it, and I have no idea, why.

Have also configured the ***@***.*** and razor.socket as described on
https://rspamd.com/doc/modules/external_services.html#razor-specific-details

 

If your razor is running, how did you configured razor itself?

 

Regards

Ralf

 

 

Von: segmentierungsfehler ***@***.***> 
Gesendet: Dienstag, 30. August 2022 22:27
An: rspamd/rspamd ***@***.***>
Cc: feiaweng2 ***@***.***>; Author ***@***.***>
Betreff: Re: [rspamd/rspamd] rspamd/razor does not work (Discussion #4237)

 

Hi

Did you already solve your problem? If not, how did you configure rspamd to
use razor?
Here is my razor part of /etc/rspamd/local.d/external_services.conf (rspamd
3.2):
razor { symbol="RAZOR" servers = "127.0.0.1:11342" timeout = 10.0
cache_expire = 7200 retransmits = 1 }
And on localhost tcp port 11342 there is razor waiting for a connection.

Bye

-
Reply to this email directly, view it on GitHub
<https://github.com/rspamd/rspamd/discussions/4237#discussioncomment-3512308
> , or unsubscribe
<https://github.com/notifications/unsubscribe-auth/A2MJ7H2OQT3HMGTH7GKWN2DV3
ZVCNANCNFSM55VPEFFA> .
You are receiving this because you authored the thread.
<https://github.com/notifications/beacon/A2MJ7H4IKFXQYUCXAHYUNO3V3ZVCNA5CNFS
M55VPEFFKYY3PNVWWK3TUL52HS4DFWFCGS43DOVZXG2LPNZBW63LNMVXHJKTDN5WW2ZLOORPWSZG
OAA2ZP5A.gif> Message ID:
***@***.***
***@***.***> >

**Comment by segmentierungsfehler** [comment]
Hi

I use the standard configuration. Is razor-check listening on localhost tcp port 11342 ? You may check on linux with:
`netstat -anp | grep 11342`
This should print something like:
`tcp        0      0 127.0.0.1:11342         0.0.0.0:*               LISTEN `

If the result is empty, you may need to setup the systemd services for razor from 
https://rspamd.com/doc/modules/external_services.html#razor-specific-details
or  you may start a listening razor-check with this command line (on linux) for testing:
`socat -t 10 TCP-LISTEN:11342,bind=127.0.0.1,fork SYSTEM:"/usr/bin/razor-check && printf spam || printf ham"`
and wait for an incoming email. 

Bye
